homeassistant/components/screenlogic/const.py

```diff
@@ -9,17 +21,45 @@
 
 SERVICE_SET_COLOR_MODE = "set_color_mode"
 ATTR_COLOR_MODE = "color_mode"
-SUPPORTED_COLOR_MODES = {
-    slugify(name): num for num, name in COLOR_MODE.NAME_FOR_NUM.items()
-}
+SUPPORTED_COLOR_MODES = {slugify(cm.name): cm.value for cm in COLOR_MODE}
 
 LIGHT_CIRCUIT_FUNCTIONS = {
-    CIRCUIT_FUNCTION.COLOR_WHEEL,
-    CIRCUIT_FUNCTION.DIMMER,
-    CIRCUIT_FUNCTION.INTELLIBRITE,
-    CIRCUIT_FUNCTION.LIGHT,
-    CIRCUIT_FUNCTION.MAGICSTREAM,
-    CIRCUIT_FUNCTION.PHOTONGEN,
-    CIRCUIT_FUNCTION.SAL_LIGHT,
-    CIRCUIT_FUNCTION.SAM_LIGHT,
+    FUNCTION.COLOR_WHEEL,
+    FUNCTION.DIMMER,
+    FUNCTION.INTELLIBRITE,
+    FUNCTION.LIGHT,
+    FUNCTION.MAGICSTREAM,
+    FUNCTION.PHOTONGEN,
+    FUNCTION.SAL_LIGHT,
+    FUNCTION.SAM_LIGHT,
+}
+
+SL_UNIT_TO_HA_UNIT = {
+    UNIT.CELSIUS: UnitOfTemperature.CELSIUS,
+    UNIT.FAHRENHEIT: UnitOfTemperature.FAHRENHEIT,
+    UNIT.MILLIVOLT: UnitOfElectricPotential.MILLIVOLT,
+    UNIT.WATT: UnitOfPower.WATT,
+    UNIT.HOUR: UnitOfTime.HOURS,
+    UNIT.SECOND: UnitOfTime.SECONDS,
+    UNIT.REVOLUTIONS_PER_MINUTE: REVOLUTIONS_PER_MINUTE,
+    UNIT.PARTS_PER_MILLION: CONCENTRATION_PARTS_PER_MILLION,
+    UNIT.PERCENT: PERCENTAGE,
 }
+
+
+def generate_unique_id(
+    device: str | None, group: int | str | None, data_key: str
+) -> str:
+    """Generate new unique_id for a screenlogic entity from specified parameters."""
+    if data_key in SHARED_VALUES and device is not None:
+        return (
+            f"{device}_{group}_{data_key}"
+            if group is not None and (isinstance(group, int) or group.isdigit())
+            else f"{device}_{data_key}"
+        )
+    return data_key
+
+
+def generate_entity_id(domain: str, device_name: str, entity_name: str) -> str:
+    """Generate new entity_id for a screenlogic entity from specified parameters."""
+    return f"{domain}.{device_name}_{entity_name}"
```

Comment:
Usually we let HA internals pick the entity id

Reply:
This is actually unused. I will remove.